### PR TITLE
[WIP] Added: Command palette component

### PR DESCRIPTION
**Pull request by rishimohan**
## What kind of change does this PR introduce?

Fixes https://github.com/supabase/supabase/issues/6986, by adding a command palette or Cmd + K search component.

## What is the new behavior?

Command Palette component can be quickly used to search through the docs, or pages from studio.

## Additional context

The PR is currently WIP. Checklist:

- [ ]  Add storybook prop documentation for the Command Palette
- [ ]  Update search results in palette to highlight matching terms

Screenshots:

image

image

### CODE REVIEW

File: src/components/CommandPalette/CommandPalette.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        >
+          <div className="px-2">
+            <IconSearch size={18} stroke="#999" strokeWidth="1.8" />
+          </div>
+          <input
+            ref={(el) => (inputRef.current = el)}
+            onKeyDown={(e) => handleShortcuts(e)}
+            value={term}
+            onChange={onSearch}
+            className="text-sm md:text-base pr-3 py-3 w-full focus:outline-none bg-transparent placeholder-dark-300 dark:placeholder-dark-400 dark:text-white"
```
Comment (MildTomato) [inline src/components/CommandPalette/CommandPalette.js:82]: @rishimohan You're probably going to have to move all the CSS classes into the theme file.

Comment (MildTomato) [reply src/components/CommandPalette/CommandPalette.js:76]: using this defaultTheme

https://github.com/supabase/ui/blob/alpha/src/lib/theme/defaultTheme.ts

Comment (MildTomato) [reply src/components/CommandPalette/CommandPalette.js:76]: then import classes like this
https://github.com/supabase/ui/blob/87776677af000bed91998403b8d3f936b8b10592/src/components/Button/Button.tsx#L88

File: src/components/CommandPalette/CommandPalette.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            placeholder="Type to search..."
+            style={{ margin: 0 }}
+          />
+          {results?.list?.length > 0 ? (
+            <div className="text-gray-800 px-4 text-sm whitespace-nowrap dark:text-dark-400">
```
Comment (MildTomato) [inline src/components/CommandPalette/CommandPalette.js:81]: everywhere you've used 'gray' you'll need to replace with 'scale' as well, so when the theme colors swap, the new colors are used.

**Review by MildTomato — CHANGES_REQUESTED** [review]

**Comment by rishimohan** [comment]
@MildTomato Thanks for the feedback. I've moved the styles to CSS file, also updated colors to use `scale` values. Understanding new theme and colors was slightly confusing for me so there's possibility I might have missed something 😅

**Comment by MildTomato** [comment]
### Supabase UI is being deprecated.

As part of that, we are closing Pull Requests in this repo for now.

If you'd like to still make these changes, you could consider making these changes in https://github.com/supabase/supabase/tree/master/packages/ui

---

We are moving the components to the main mono repo at [https://github.com/supabase/supabase](https://github.com/supabase/supabase).

The auth component has been moved to it's own repo and package. [https://github.com/supabase-community/auth-ui](https://github.com/supabase-community/auth-ui). You can also read the [docs for auth ui here](https://supabase.com/docs/guides/auth/auth-helpers/auth-ui).

**Pull request closed without merging**